src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -128,23 +129,28 @@ def pex_requirements(
         If the tool supports lockfiles, the returned type will install from the lockfile rather than
         `all_requirements`.
         """
-        if not self.register_lockfile or self.lockfile == "<none>":
+
+        if not self.is_lockfile:
             return PexRequirements((*self.all_requirements, *extra_requirements))
+
+        requirements = FrozenOrderedSet([*self.all_requirements, *extra_requirements])
```

Comment:
Some duplication here. You could update `calculate_invalidation_digest` to take `Iterable[str]` instead of `FrozenOrderedSet[str]`, which would allow you to define the `PexRequirements` once and then pass `requirements.req_strings`(?) to the function.